core/go/internal/sequencer/originator/transaction/state_machine.go

```diff
@@ -0,0 +1,586 @@
+					Transitions: []Transition{
+						{
+							To: State_Delegated,
+						},
+					},
+				},
+				Event_AssembleRequestReceived: {
+					// For some reason we've been asked to assemble again. We must not have moved to endorsement gathering,
+					// reverted, or parked. This could be because of a temporary issue preventing assembly (e.g. we couldn't
+					// resolve a remote verifier while it was offline). Assuming this is a new request, action it.
+					Validator: validator_AssembleRequestMatches,
+					Actions: []ActionRule{
+						{
+							If:     guard_Not(guard_AssembleRequestMatchesPreviousResponse),
+							Action: action_AssembleAndSign,
+						},
+						{
+							If:     guard_AssembleRequestMatchesPreviousResponse,
+							Action: action_ResendAssembleSuccessResponse,
+						},
+					},
+					// No transition - we're still assembling
+				},
+			},
+		},
+		State_Endorsement_Gathering: {
+			Events: map[EventType]EventHandler{
+				Event_AssembleRequestReceived: {
+					Validator: validator_AssembleRequestMatches,
+					Actions: []ActionRule{
+						{
+							//We thought we had got as far as endorsement but it seems like the coordinator had not got the response in time and has resent the assemble request, we simply reply with the same response as before
+							If:     guard_AssembleRequestMatchesPreviousResponse,
+							Action: action_ResendAssembleSuccessResponse,
+						}},
+					Transitions: []Transition{{
+						//This is different from the previous request. The coordinator must have decided that it was necessary to re-assemble with different available states so we go back to assembling state for a do-over
+						If: guard_Not(guard_AssembleRequestMatchesPreviousResponse),
+						To: State_Assembling,
+					}},
+				},
+				Event_CoordinatorChanged: {
+					Transitions: []Transition{
+						{
+							To: State_Delegated,
+						},
+					},
+				},
+				Event_PreDispatchRequestReceived: {
+					Validator: validator_PreDispatchRequestMatchesAssembledDelegation,
+					Transitions: []Transition{
+						{
+							To: State_Prepared,
+							On: action_SendPreDispatchResponse,
+						},
+					},
+				},
+			},
+		},
+		State_Prepared: {
+			Events: map[EventType]EventHandler{
+				Event_Dispatched: {
+					//Note: no validator here although this event may or may not match the most recent dispatch confirmation response.
+					// It is possible that we timed out  on Prepared state, delegated to another coordinator, got as far as prepared again and now just learning that
+					// the original coordinator has dispatched the transaction.
+					// We can't do anything to stop that, but it is interesting to apply the information from event to our state machine because we don't know which of
+					// the many base ledger transactions will eventually be confirmed and we are actually not too fussy about which one does
```

Comment:
I made the comment on one of the other state machines, that this is really critical stuff these comments. It's protocol specification stuff, not code.

I think these comments are great examples of where we should maintain the comments in the spec along with the state diagram, and refer to each line in the graph by a cross-reference. Maybe even force that.

We could get clever and have the comments in the code, to generate the docs - but that might be a step further than we need/can-afford here.